############################
User guide
############################


****************************
pairwisedist module
****************************

*pairwisedist* can calculate the pairwise-distance matrix for an array of *n* samples by *p* features.
The distance metric currently supported by *pairwisedist* are the `Son and Baek dissimilarities <https://doi.org/10.1016/j.patrec.2007.09.015>`_ *YS1* and *YR1*.

All functions in *pairwisedist* receive as input an n-by-p matrix.
If the the matrix consists of row variables (parameter 'rowvar' == True - the default), *pairwisedist* will return an n-by-n pairwise distance matrix. Otherwise, the pairwise distance will be calculated on the columns, and *pairwisedist* will return a p-by-p pairwise distance matrix.

The YS1 and YR1 dissimilarity metrics
======================================

The Son and Baek dissimilarity metrics, also know as YS1 and YR1, were developed as an improved correlation metric for time-series Gene Expression data, generated by Microarray or RNA Sequencing experiments.
The goal of these distance metrics is to capture more accurately the shape of the expression pattern of each gene, and rank genes with similar expression patterns as more similar to one another.
You can read more about these metrics in `The original publication <https://doi.org/10.1016/j.patrec.2007.09.015>`_.

In principle, the YS1 and YR1 scores integrate information about the the correlation between the different samples (Pearson correlation (R) for YR1 metric, Spearman correlation (S) for YS1 metric),
the positon of the minimal and maximal values of each sample (M) and the agreement of their slopes (A)s.
The weight of each of the components can be determined by the user using the parameters 'omega1', 'omega2' and 'omega3' respectively.
The default weights recommended by the authors are 0.5, 0.25, 0.25.

Due to the characteristics of the YS1 and YR2 metrics, the order of the features (columns) in the input matrix affect the values of the pairwise distance. 